aiida_restapi/routers/graphql/utils.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+import datetime
+from typing import Any, Dict, Iterator, List, Type
+
+import graphene as gr
+from aiida import orm
+from aiida.cmdline.utils.decorators import with_dbenv
+from dateutil.parser import parser as date_parser
+from graphene.types.generic import GenericScalar
+from graphql import GraphQLError
+from graphql.language import ast
+
+ENTITY_LIMIT = 100
+"""The maximum query limit allowed for a list of entities."""
+
+
+class JSON(GenericScalar):
+    """
+    Subclass of the  `GenericScalar` scalar type represents a generic
+    GraphQL scalar value that could be:
+    String, Boolean, Int, Float, List or Object.
+    """
+
+
+def make_entities_cls(
+    entity_cls: Type[gr.ObjectType], orm_cls: Type[orm.Entity], name: str
+) -> Type[gr.ObjectType]:
+    """Return a class with standard fields/resolvers for querying multiple rows of the same entity."""
+
+    # TODO is this the best way to achieve this?
```

Comment:
I'm not 100% sure whether that is what you're looking for here - just checking whether you're aware of the option of subclassing from `Generic` and thereby introducing a template variable into the class itself (not just the typing) as done e.g. [here](https://github.com/aiidateam/aiida-restapi/pull/18/files#diff-c4740058b2de03bdf40c5e266dda1d092ae5ca2eda36ec57df85af6518c0eef9R33)

e.g. something like
```
OrmT = TypeVar("OrmT", bind=...)

class Entities(gr.ObjectType, Generic[OrmT]):
   ...
   orm.QueryBuilder().append(OrmT, filters=filters) ...

# later instantiate template with
Entities[orm.User]
```

Reply:
oh yeh thanks, I'm semi-uptodate with all this templating business. Haven't gone through yet and properly addressed all the typing and linting, etc

Reply:
Actually this might be the canonical way to deal with such subclassing: https://stackoverflow.com/a/56570594/5033292, using `class Meta:` as all graphene-pydantic/django/sqlalchemy do